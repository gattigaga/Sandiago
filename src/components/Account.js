import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { TouchableNativeFeedback } from "react-native";

const Container = styled.View`
  padding: 24px;
  border-bottom-width: 0.5;
  border-bottom-color: rgba(255, 255, 255, 0.2);
`;

const Name = styled.Text`
  font-family: Roboto-Bold;
  font-size: 18px;
  color: white;
  margin-bottom: 8px;
`;

const Description = styled.Text`
  font-family: Roboto-Regular;
  font-size: 14px;
  color: white;
`;

const Date = styled.Text`
  font-family: Roboto-Regular;
  font-size: 12px;
  color: rgba(255, 255, 255, 0.5);
  text-align: right;
  margin-bottom: 8px;
`;

const Account = ({ name, description, date, onPress }) => (
  <TouchableNativeFeedback onPress={onPress}>
    <Container>
      <Date>{date}</Date>
      <Name>{name}</Name>
      <Description>{description}</Description>
    </Container>
  </TouchableNativeFeedback>
);

Account.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  date: PropTypes.string,
  onPress: PropTypes.func
};

export default Account;
