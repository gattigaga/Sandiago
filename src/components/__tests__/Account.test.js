import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Account from "../Account";

describe("Account", () => {
  const setup = propOverrides => {
    const props = {
      name: "Facebook",
      description: "This is my first Facebook account",
      date: "09:05 AM",
      onPress: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Account {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onPress' while pressed", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("press");

    expect(props.onPress).toBeCalled();
  });
});
