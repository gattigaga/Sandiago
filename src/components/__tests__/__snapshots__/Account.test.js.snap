// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should renders default 1`] = `
<DummyTouchableNativeFeedback
  onPress={[MockFunction]}
>
  <Styled(Component)>
    <Styled(Text)>
      09:05 AM
    </Styled(Text)>
    <Styled(Text)>
      Facebook
    </Styled(Text)>
    <Styled(Text)>
      This is my first Facebook account
    </Styled(Text)>
  </Styled(Component)>
</DummyTouchableNativeFeedback>
`;
